File: Algorithms/189. Rotate Array/swift/solution.extra_array.swift
class Solution {
    func rotate(_ nums: inout [Int], _ k: Int) {
        let n = nums.count
        var a: [Int] = Array.init(repeating: 0, count: n)
        for i in 0..<n {
            a[(i + k) % n] = nums[i]
        }
        for i in 0..<n {
            nums[i] = a[i]
        }
    }
}

File: Algorithms/189. Rotate Array/php/solution.extra_array.php
<?php
class Solution
{

    /**
     * @param Integer[] $nums
     * @param Integer $k
     * @return NULL
     */
    function rotate(&$nums, $k)
    {
        $n = count($nums);
        $a = array_fill(0, $n, 0);
        for ($i = 0; $i < $n; $i++) {
            $a[($i + $k) % $n] = $nums[$i];
        }
        for ($i = 0; $i < $n; $i++) {
            $nums[$i] = $a[$i];
        }
    }
}
